db/sql/00_msar.sql

```diff
@@ -2135,16 +2135,21 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
 CREATE OR REPLACE FUNCTION
-msar.drop_table(tab_id oid, cascade_ boolean, if_exists boolean) RETURNS text AS $$/*
-Drop a table, returning the command executed.
+msar.drop_table(tab_id oid, cascade_ boolean) RETURNS text AS $$/*
+Drop a table, returning the fully qualified name of the dropped table.
 
 Args:
   tab_id: The OID of the table to drop
   cascade_: Whether to drop dependent objects.
-  if_exists_: Whether to ignore an error if the table doesn't exist
 */
+DECLARE relation_name text;
 BEGIN
-  RETURN __msar.drop_table(__msar.get_relation_name(tab_id), cascade_, if_exists);
+  relation_name := __msar.get_relation_name(tab_id);
+  -- if_exists doesn't work while working with oids because 
+  -- the SQL query gets parameterized with tab_id instead of relation_name
+  -- since we're unable to find the relation_name for a non existing table. 
+  PERFORM __msar.drop_table(relation_name, cascade_, if_exists => false);
+  RETURN relation_name;
```

Comment:
I am okay with returning `NULL` here instead of the table name. There are a couple of reasons why I chose to make this change.
- Our `columns.delete` function returns the number of columns deleted. 
- Earlier this function used to return the SQL query that was being run on the db for dropping the table.

Keeping this in mind I thought if an end user were to use our RPC endpoint, it might be beneficial for them if they receive some kind of a not null response.

Reply:
I prefer returning either `1` for "1 table dropped", or the name, is preferable:
- Even if we're not displaying on the FE, it's useful for a user just calling the endpoint.
- It distinguishes between the "no table found" and "found and dropped table" cases, whenever `if_exists=true`

Reply:
I thought we agreed yesterday that we would not bother with using parameters for `if_exists`. Instead we would use queries that drop things without `IF EXISTS` (raising an exception if the thing is not found) and add tables without `IF NOT EXISTS` (raising an exception if the thing is already there). That's simpler.

> it's useful for a user just calling the endpoint

My point here is that we don't currently have any use cases like this. And I don't foresee having any use cases like this. If we do end up having these use cases then we can add this complexity at that point.

Returning void is simpler. Changing a void return to a value return is a non-breaking change. Easy to make later on! Changing a value return to a different return is a breaking change. Harder to make later on! I want to start simple and then grow the complexity only as needed.